misc/python/materialize/cli/mzcompose.py

```diff
@@ -625,7 +626,16 @@ def handle_composition(
             # test analytics, even if the workflow doesn't define more granular
             # test cases.
             with composition.test_case(f"workflow-{args.workflow}"):
-                composition.workflow(args.workflow, *args.unknown_subargs[1:])
+                ci_extra_args = json.loads(os.getenv("CI_EXTRA_ARGS", "{}"))
+                buildkite_step_key = os.getenv("BUILDKITE_STEP_KEY")
+                extra_args = (
+                    ci_extra_args[buildkite_step_key].split(" ")
+                    if buildkite_step_key
+                    else []
+                )
+                composition.workflow(
+                    args.workflow, *args.unknown_subargs[1:], *extra_args
```

Comment:
Do the `extra_args` override existing args if they are already set?

Reply:
Depends on how Python interprets them. We just append them after the normal args. You can use argparse to overwrite with the last value, or interpret them as a list.

Reply:
Should we use `setattr(args, param, new_value)`?